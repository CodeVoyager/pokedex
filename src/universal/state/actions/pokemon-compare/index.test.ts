/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  pushPokemonCompareCandidateAction,
  setPokemonCompareCurrentAction,
  clearPokemonCompareCandidatesAction,
  clearPokemonCompareCurrentAction,
} from '.';

describe('State::Actions', () => {
  describe('PokemonCompare', () => {
    describe('pushPokemonCompareCandidateAction creator', () => {
      test('should return correct action', () => {
        expect(
          pushPokemonCompareCandidateAction({
            foo: 'bar',
          } as any)
        ).toMatchSnapshot();
      });
    });
    describe('setPokemonCompareCurrentAction creator', () => {
      test('should return correct action', () => {
        expect(
          setPokemonCompareCurrentAction({
            foo: 'bar',
          } as any)
        ).toMatchSnapshot();
      });
    });
    describe('clearPokemonCompareCandidatesAction creator', () => {
      test('should return correct action', () => {
        expect(clearPokemonCompareCandidatesAction()).toMatchSnapshot();
      });
    });
    describe('clearPokemonCompareCurrentAction creator', () => {
      test('should return correct action', () => {
        expect(clearPokemonCompareCurrentAction()).toMatchSnapshot();
      });
    });
  });
});
