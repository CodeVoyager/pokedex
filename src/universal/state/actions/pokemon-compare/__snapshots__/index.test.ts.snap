// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State::Actions PokemonCompare clearPokemonCompareCandidatesAction creator should return correct action 1`] = `
Object {
  "type": "CLEAR_POKEMON_COMPARE_CURRENT",
}
`;

exports[`State::Actions PokemonCompare clearPokemonCompareCurrentAction creator should return correct action 1`] = `
Object {
  "type": "CLEAR_POKEMON_COMPARE_CURRENT",
}
`;

exports[`State::Actions PokemonCompare pushPokemonCompareCandidateAction creator should return correct action 1`] = `
Object {
  "payload": Object {
    "foo": "bar",
  },
  "type": "PUSH_POKEMON_COMPARE_CANDIDATE",
}
`;

exports[`State::Actions PokemonCompare setPokemonCompareCurrentAction creator should return correct action 1`] = `
Object {
  "payload": Object {
    "foo": "bar",
  },
  "type": "SET_POKEMON_COMPARE_CURRENT",
}
`;
